core/edr-core/src/main/java/org/eclipse/tractusx/edc/edr/core/service/EdrServiceImpl.java

```diff
@@ -15,48 +15,40 @@
 package org.eclipse.tractusx.edc.edr.core.service;
 
 import org.eclipse.edc.connector.contract.spi.types.negotiation.ContractNegotiation;
-import org.eclipse.edc.connector.contract.spi.types.negotiation.ContractRequest;
-import org.eclipse.edc.connector.contract.spi.types.negotiation.ContractRequestData;
-import org.eclipse.edc.connector.spi.contractnegotiation.ContractNegotiationService;
 import org.eclipse.edc.service.spi.result.ServiceResult;
 import org.eclipse.edc.spi.query.QuerySpec;
-import org.eclipse.edc.spi.types.domain.callback.CallbackAddress;
 import org.eclipse.edc.spi.types.domain.edr.EndpointDataReference;
+import org.eclipse.tractusx.edc.edr.spi.EdrManager;
 import org.eclipse.tractusx.edc.edr.spi.service.EdrService;
 import org.eclipse.tractusx.edc.edr.spi.store.EndpointDataReferenceCache;
 import org.eclipse.tractusx.edc.edr.spi.types.EndpointDataReferenceEntry;
 import org.eclipse.tractusx.edc.edr.spi.types.NegotiateEdrRequest;
 
 import java.util.List;
 import java.util.Optional;
-import java.util.Set;
 import java.util.stream.Collectors;
-import java.util.stream.Stream;
 
 import static java.lang.String.format;
 
 public class EdrServiceImpl implements EdrService {
 
-    public static final String LOCAL_ADAPTER_URI = "local://adapter";
-    public static final Set<String> LOCAL_EVENTS = Set.of("contract.negotiation", "transfer.process");
-    public static final CallbackAddress LOCAL_CALLBACK = CallbackAddress.Builder.newInstance()
-            .transactional(true)
-            .uri(LOCAL_ADAPTER_URI)
-            .events(LOCAL_EVENTS)
-            .build();
-    private final ContractNegotiationService contractNegotiationService;
+    private final EdrManager edrManager;
 
     private final EndpointDataReferenceCache endpointDataReferenceCache;
 
-    public EdrServiceImpl(ContractNegotiationService contractNegotiationService, EndpointDataReferenceCache endpointDataReferenceCache) {
-        this.contractNegotiationService = contractNegotiationService;
+    public EdrServiceImpl(EdrManager edrManager, EndpointDataReferenceCache endpointDataReferenceCache) {
+        this.edrManager = edrManager;
         this.endpointDataReferenceCache = endpointDataReferenceCache;
     }
 
     @Override
     public ServiceResult<ContractNegotiation> initiateEdrNegotiation(NegotiateEdrRequest request) {
-        var contractNegotiation = contractNegotiationService.initiateNegotiation(createContractRequest(request));
-        return ServiceResult.success(contractNegotiation);
+        var contractNegotiation = edrManager.initiateEdrNegotiation(request);
```

Comment:
Lately, I'thinking that the "sync" operations should be made directly in the service without passing through the manager that should only be the running state machine. In fact also in the `ContractNegotiation` and `TransferProcess` services, the managers are only used to initiate the processes, but I think I will inline them the next time I will pass there, or switch them to command/command handler. At that point the "*Manager" could be called "*StateMachine". (just raving around a little :grin: )

Reply:
Will open an issue about this. In fact it was inlined previously :)